Feature/vertically aligned search container

CorreyL opened the pull request:
## Preamble

This PR concerns right-justifying the `SearchContainer` component.

This PR fixes Issue #4

## Demo

![search-container-vertically-aligned](https://user-images.githubusercontent.com/16601729/85159380-4a10ec80-b212-11ea-8a40-5b0025688ba6.gif)

For reference, this is what it looked like before:

![search-prev-next](https://user-images.githubusercontent.com/16601729/85079986-ef798100-b17c-11ea-9f97-e490db9760ed.gif)

CorreyL committed:
Render SearchContainer on right-side of the app

- Refactor the opening tag in the `SearchContainer` component from a
  `div` to a `span` to ensure the element acts as an `inline` element,
  rather than a `block` element
- Move the usage of the `SearchContainer` element inside of the `div`
  where the toolbar buttons are so that the `SearchContainer` component
  is rendered starting at the top-right corner of the app
- Give the opening `span` element in `SearchContainer` an `id` of
  `search-container` and apply `float` to it so that rendering of the
  component begins on the right side of the page
- Uncomment previously implemented logic that adjusted the width of the
  `SearchContainer` and `Webviewer` elements, now that `SearchContainer`
  renders starting at the top-right of the app as expected

CorreyL committed:
Render mode toggles on separate line

Ensures word wrapping does not occur on the text for the `input`
`checkbox` elements

CorreyL committed:
Move search related buttons below checkbox inputs

Ensures consistency with the latest release of the Webviewer beta UI

Rendering the buttons on their own block also ensures smaller
resolutions are supported without line-wrapping occurring

CorreyL commented on src/components/SearchContainer/SearchContainer.css:
```diff
@@ -0,0 +1,3 @@
+#search-container {
+  float: right;
```
Let me know if you want me to take the time to implement a proper [`Flexbox` solution](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)

andreysaf reviewed (commented):
Good so far, I think proper flexbox plus apply the same CSS as the new UI, same colours for buttons to show users that they can achieve the exact look and feel as our UI.

CorreyL committed:
Render SearchContainer using flexbox

Rather than performing a hacky use of the `float` attribute to render
the `SearchContainer` component to the right of the PDF viewer, define a
CSS class that makes all immediate children elements a `flexbox`
container, thus correctly rendering the `SearchContainer` component to
the right of the PDF viewer

Move the use of `SearchContainer` as a sibling element of the
`div#viewer` element, ensuring that the `.flexbox-container` class will
be applied to both elements

Remove all logic in a `useEffect` that previously adjusted the width of
the components that supported the previously used `float` solution

CorreyL commented:
> Good so far, I think proper flexbox [...]

Addressed in 330b527

> plus apply the same CSS as the new UI, same colours for buttons to show users that they can achieve the exact look and feel as our UI.

Working on this next

CorreyL commented:
> plus apply the same CSS as the new UI, same colours for buttons to show users that they can achieve the exact look and feel as our UI.

Implementing in [feature/search-results-list](https://github.com/PDFTron/webviewer-custom-ui/commits/feature/search-results-list) to minimize merge conflicts.

andreysaf commented on src/App.js:
```diff
@@ -136,16 +96,16 @@ const App = () => {
-        open={searchContainerOpen}
-      />
-      <div id="scroll-view" ref={scrollView}>
+      <div class="flexbox-container" id="scroll-view" ref={scrollView}>
```
class -> className

CorreyL replied:
Haha good eye, already fixed in my new branch in https://github.com/PDFTron/webviewer-custom-ui/commit/7b8b8afde985d504160469040ab32d1652ee6bcc 👍

andreysaf reviewed (approved):
Few things that can be resolved in future PRs:

- When clicking search, auto focus on search input
- The document scrolls by clicking next with the search container
